query/src/sql/planner/binder/sort.rs

```diff
@@ -13,28 +13,26 @@
 // limitations under the License.
 
 use common_ast::ast::OrderByExpr;
-use common_datavalues::DataTypeImpl;
 use common_exception::Result;
 
 use crate::sql::binder::scalar::ScalarBinder;
 use crate::sql::binder::Binder;
 use crate::sql::optimizer::SExpr;
-use crate::sql::plans::Scalar;
 use crate::sql::plans::SortItem;
 use crate::sql::plans::SortPlan;
 use crate::sql::BindContext;
 
 impl Binder {
-    pub(super) fn bind_order_by(
+    pub(super) async fn bind_order_by(
         &mut self,
         order_by: &[OrderByExpr],
         bind_context: &mut BindContext,
     ) -> Result<()> {
-        let scalar_binder = ScalarBinder::new(bind_context);
-        let order_by_exprs = order_by
-            .iter()
-            .map(|expr| scalar_binder.bind_expr(&expr.expr))
-            .collect::<Result<Vec<(Scalar, DataTypeImpl)>>>()?;
+        let scalar_binder = ScalarBinder::new(bind_context, self.ctx.clone());
+        let mut order_by_exprs = vec![];
+        for expr in order_by {
+            order_by_exprs.push(scalar_binder.bind_expr(&expr.expr).await?);
```

Comment:
You may consider using https://docs.rs/futures/latest/futures/prelude/stream/struct.FuturesOrdered.html